sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/AppManagedUserAttributeController.java

```diff
@@ -0,0 +1,323 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@Tag(
+    name = "Manage user's attribute for app",
+    description = "New endpoints for apps to get add and delete some attributes on users")
+@RequestMapping(value = {"/v2", "/"})
+@SecurityRequirements(
+    value = {@SecurityRequirement(name = "oAuth"), @SecurityRequirement(name = "basic")})
+public class AppManagedUserAttributeController {
+
+  @Autowired private UserService userService;
+
+  @Autowired private PermissionService permissionService;
+
+  @Autowired private RealmProvider realmProvider;
+
+  @PatchMapping(
+      value = {
+        "/realms/{realm}/storages/{storage}/users/{id}/{app-managed-attribute-name}/{app-managed-attribute-value}"
+      },
+      consumes = {MediaType.APPLICATION_JSON_VALUE},
+      produces = {MediaType.APPLICATION_JSON_VALUE})
+  @Operation(summary = "add attribute managed by application")
+  @PreAuthorize("@NewAuthorizeMethodDecider.isReader(#realm,#storage)")
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "User updated",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = User.class))
+            }),
+        @ApiResponse(
+            responseCode = "404",
+            description = "User not found",
+            content = {@Content(mediaType = "application/json")}),
+        @ApiResponse(
+            responseCode = "400",
+            description = "Invalid combinaison of id between body and path",
+            content = {@Content(mediaType = "application/json")})
+      })
+  public ResponseEntity<?> addUserAttributesManagedByApp(
+      @Parameter(
+              description = "Name of the realm where the operation will be made",
+              required = true)
+          @PathVariable("realm")
+          String realm,
+      @Parameter(
+              description = "Name of the userstorage where the operation will be made",
+              required = true)
+          @PathVariable("storage")
+          String storage,
+      @Parameter(description = "User's id to update", required = true) @PathVariable("id")
+          String id,
+      @Parameter(description = "key of attribute to add", required = true)
+          @PathVariable("app-managed-attribute-name")
+          String attributeKey,
+      @Parameter(description = "value of attribute to add", required = true)
+          @PathVariable("app-managed-attribute-value")
+          String attributeValue) {
+
+    Realm _realm = realmProvider.load(realm);
+    if (attributeKey
+        .toUpperCase()
+        .equals(
+            _realm.getProperties().get(GlobalKeysConfig.APP_MANAGED_ATTRIBUTE_KEY).toUpperCase())) {
```

Comment:
Here we don't use the default value but only the value in the realm if set. Should the realm be loaded with a default ? Should we check for existence here and use default if null ? What should happen if neither is set ?

Reply:
when the realm is loaded we check in the provider if the value is set, if none we use default none set in properties. If both are not set so we have a comparison between a string and a null value so it will fail and the user will receive an 401 error code